canister/src/unstable_blocks.rs

```diff
@@ -71,6 +71,19 @@ impl UnstableBlocks {
         self.network
     }
 
+    /// Returns all blocks in the tree with the number of confirmations
+    /// separated by heights.
+    pub fn blocks_with_confirmations_by_height(
+        &self,
+        track_until_height: usize,
```

Comment:
Do we need this `track_until_height` parameter? In practice, it's almost always the case that the main chain is the longest chain, so we'd need to go through the entire `BlockTree`. I'd suggest removing this parameter to simplify the method.

Reply:
Done :)